### PR TITLE
Find Poco library using upstream CMake configuration.

### LINKED ISSUE #117: linking tests fails with CMAKE_BUILD_TYPE=Debug
Without specifying the build type `Poco_LIBRARIES` contains `/usr/lib/libPocoFoundation.so;dl`. When building debug the variable only contains `dl` - I assume because no debug libraries are available on the system.

As a result building the tests fails with unresolved symbols:

```
../devel/lib/libclass_loader.so: undefined reference to `typeinfo for Poco::RuntimeException'
../devel/lib/libclass_loader.so: undefined reference to `Poco::SharedLibrary::isLoaded() const'
../devel/lib/libclass_loader.so: undefined reference to `Poco::SharedLibrary::suffix[abi:cxx11]()'
../devel/lib/libclass_loader.so: undefined reference to `typeinfo for Poco::LibraryLoadException'
../devel/lib/libclass_loader.so: undefined reference to `typeinfo for Poco::LibraryAlreadyLoadedException'
../devel/lib/libclass_loader.so: undefined reference to `Poco::SharedLibrary::SharedLibrary(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../devel/lib/libclass_loader.so: undefined reference to `typeinfo for Poco::NotFoundException'
../devel/lib/libclass_loader.so: undefined reference to `Poco::SharedLibrary::unload()'
```

### PR DESCRIPTION
**Pull request by nuclearsandwich**
A bug in the FindPoco module from cmake_modules causes incorrect linking in Debug configurations.

I haven't decided if this is a straw man or not since I'm not sure what we gain or lose by not using the module from cmake_modules.

This is one possible way to resolve #117 but I'd like input from someone like @wjwwood before I decide whether this is a good idea or not.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,6 @@ endif()
 find_package(console_bridge REQUIRED)
 
 if(${catkin_FOUND})
-  find_package(catkin REQUIRED COMPONENTS cmake_modules)
```
Comment (dirk-thomas) [inline CMakeLists.txt:22]: The dependency on `cmake_modules` should then be removed from the manifest.

**Comment by nuclearsandwich** [comment]
CI informs me that this won't be a complete solution as Debian Stretch's libpoco-dev package doesn't contain CMake configuration.

**Comment by wjwwood** [comment]
+1 to getting rid of stuff in `cmake_modules` where ever possible, but as you said, we likely created it there to address a specific problem like debian not shipping proper CMake module or config file.

**Comment by nuclearsandwich** [comment]
While this would be the preferred solution to #117 it won't work for Melodic. I've added a commit that completes this PR so it can be considered in the future once support for Stretch is dropped.

**Pull request closed without merging**